client/src/main/java/org/threadly/litesockets/client/http/HTTPClient.java

```diff
@@ -589,11 +589,8 @@ public void onClose(Client client) {
 
       client.close();
       if(hrw != null) {
-        boolean wasProcessing = hrw.hrp.isProcessing();
         hrw.hrp.connectionClosed();
-        if(! hrw.slf.isDone() && ! wasProcessing) {
-          process(hrw);
-        } else {
+        if(! hrw.slf.isDone()) {
           hrw.slf.setFailure(new HTTPParsingException("Did not get complete body!"));
```

Comment:
Need to be careful here, if the connection type is close w/o a ContentLength, or http1.0 we might not know how large the body is until the connection is closed, and even if it closed early before it sent all the data it is not actually an error from an HTTP perspective.

Reply:
I am not sure I understand the case you're describing, I may need some help in fixing this bug.

To me this seemed safe because we are only setting the future if it is not already complete.  I can't rationalize a reason that we would get a close BEFORE the read.  So we should at this point be good to say "if you're not done, you're now in error at this point", assuming there is no retry logic which is what I assumed was being attempted here.